lib/rubocop/cop/netlify/require_scope_semantics.rb

```diff
@@ -0,0 +1,44 @@
+# frozen_string_literal: true
+
+require_relative "require_scope_base"
+
+module RuboCop
+  module Cop
+    module Netlify
+      # This cop checks OAuth scope semantic mismatches
+      #
+      # @example
+      #   # bad
+      #   require_scope "all:read"
+      #   def destroy
+      #
+      #   # good
+      #   require_scope "all:read"
+      #   def index
+      class RequireScopeSemantics < RequireScopeBase
+        def on_def(node)
+          return unless @is_controller
+          return unless @method_protection == :public
+
+          scopes = scopes_for_action(node.method_name)
+          return if scopes.empty?
+          scopes = scopes.last # this is the observed matching behavior
+
+          if ["new", "update", "create", "destroy", "delete"].any? { |s| node.method_name.to_s.include?(s) }
```

Comment:
It would be nice to extract this array to a constant like `WRITEABLE_METHODS` (I know there's a better name for this, but doesn't come to my mind now) and put that in the class.
I'd also add `edit` even though we don't use it much in our codebase.

Question on the `any? { |s| node.method_name.to_s.include?(s) }` part. That means you're trying to catch things like `def destroy_something` too, right? not just `def destroy` methods.

Reply:
Yeah. They are keywords that strongly indicate whether they are mutating operations or not.

I'll add `MUTATION_KEYWORDS` or similar. Ooh. `edit` as well. Good catch. I'll have another read through our bitballoon rails routes for other keywords too.

Reply:
Done. Thanks!